lib/astutils.cpp

```diff
@@ -1131,6 +1131,35 @@ bool isConstVarExpression(const Token *tok)
     return false;
 }
 
+static const Variable *getLHSVariableRecursive(const Token *tok)
+{
+    if (!tok)
+        return nullptr;
+    if (Token::Match(tok, "*|&|&&|[")) {
```

Comment:
If the code is `buf[10]` then the `buf` variable is returned. If the code is `*(buf + 10)` shouldn't the `buf` variable be returned? What should be returned if the code is `*(buf + i)` where buf is a pointer and i is an integer?

Reply:
It can be extended for that case as well.